### ISSUE TITLE
[SR-11546] LLDB gets Swift Foundation overlay optionals wrong

### ISSUE THREAD

**Issue by mattneub** [swiftlang/llvm-project#4479]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-11546      |
   |Radar             | None         |
   |Original Reporter | @mattneub      |
   |Type              | Bug    |
   |Status            | Resolved        |
   |Resolution        | Duplicate    |
        

Attachment: [Download](https://user-images.githubusercontent.com/2727770/164963473-8a3a644c-7b8e-4548-9c73-4a0bf0ed7436.gz)

      
<details>
  <summary>Environment</summary>

Xcode Version 11.1 (11A1027)

</details>

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 1         |
   |Component/s       | LLDB for Swift    |
   |Labels            | Bug        |
   |Assignee          | @dcci      |
   |Priority          | Medium      |

   

   md5: a862911179814104bbf69ddcd81faac5

  </details>

**blocks**:  
* [SR-11593](https://bugs.swift.org/browse/SR-11593) REPL incorrectly reports URL as nil  

**relates to**:  
* [SR-11593](https://bugs.swift.org/browse/SR-11593) REPL incorrectly reports URL as nil  

**Issue Description:**

Run this code in a real iOS app with a breakpoint at the last line:

``` java
        struct Booking {
            var start_time: Date?
        }
        let booking : Booking? = Booking()
        
        let urlAsString = "http://www.google.de/"
        let url = URL(string: urlAsString)
        
        print(booking, url) // breakpoint here
```

Look in the variables list in the debugger. Both \`booking\` and \`url\` are shown as \`nil\`. But in fact they are not \`nil\`, as the console will prove (see screen shot).

If you change Date and URL to NSDate and NSURL the problem goes away. So it is confined to Swift overlay types.

**Comment by 0acf245c-f0fa-406f-afb9-4746d10f0359** [comment]
Dupe of SR-11593
